ext/LogDensityProblemsADEnzymeExt.jl

```diff
@@ -45,8 +45,24 @@ function ADgradient(::Val{:Enzyme}, ℓ; mode::Enzyme.Mode = Enzyme.Reverse, sha
     if mode isa Enzyme.ReverseMode && shadow !== nothing
         @info "keyword argument `shadow` is ignored in reverse mode"
         shadow = nothing
+
     end
-    return EnzymeGradientLogDensity(ℓ, mode, shadow)
+    return EnzymeGradientLogDensity(ℓ, _set_needs_primal(mode), shadow)
+end
+
+function _set_needs_primal(ad::Enzyme.ReverseMode{PR, RT, ABI, Holo, Err}) where {PR, RT, ABI, Holo, Err}
+    PR && return ad
+    return Enzyme.ReverseMode{true, RT, ABI, Holo, Err}()
+end
+
+function _set_needs_primal(ad::Enzyme.ForwardMode{PR, ABI, Err, RT}) where {PR, ABI, Err, RT}
+    PR && return ad
+    return Enzyme.ForwardMode{true, ABI, Err, RT}()
+end
+
+function _set_needs_primal(ad::Enzyme.Mode) 
+    throw(ArgumentError("Enzyme mode ($ad) is not supported."*
+                        "Currently only `ReverseMode` and `ForwardMode` are supported."))
```

Comment:
There's a white space missing, it seems:
```suggestion
    throw(ArgumentError("Unsupported Enzyme mode $ad: Currently only `Enzyme.ReverseMode` and `Enzyme.ForwardMode` are supported."))
```